src/Controller/VehicleController.php

```diff
@@ -78,4 +80,87 @@ public function delete(Request $request, Vehicle $vehicle, EntityManagerInterfac
 
         return $this->redirectToRoute('vehicle_index', [], Response::HTTP_SEE_OTHER);
     }
+
+    #[Route('/{id}/parts', name: 'vehicle_parts', methods: ['GET'])]
+    public function parts(Vehicle $vehicle): Response
+    {
+        return $this->render('vehicle/parts.html.twig', [
+            'vehicle' => $vehicle,
+        ]);
+    }
+
+    #[Route('/{id}/parts/new', name: 'vehicle_parts_new', methods: ['GET', 'POST'])]
+    public function partsNew(Request $request, Vehicle $vehicle, EntityManagerInterface $entityManager): Response
+    {
+        $part = new Part();
+        $part->setVehicle($vehicle);
+        $form = $this->createForm(PartType::class, $part);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $entityManager->persist($part);
+            $entityManager->flush();
+
+            return $this->redirectToRoute('vehicle_parts', [
+                'id' => $vehicle->getId(),
+            ], Response::HTTP_SEE_OTHER);
+        }
+
+        return $this->renderForm('vehicle/parts_new.html.twig', [
+            'vehicle' => $vehicle,
+            'part' => $part,
+            'form' => $form,
+        ]);
+    }
+
+    #[Route('/{id}/parts/{partId}', name: 'vehicle_parts_show', methods: ['GET'])]
+    public function partsShow(Vehicle $vehicle, Part $part): Response
+    {
+        return $this->render('vehicle/parts_show.html.twig', [
+            'vehicle' => $vehicle,
+            'part' => $part,
+        ]);
+    }
+
+    #[Route('/{id}/parts/{partId}/edit', name: 'vehicle_parts_edit', methods: ['GET', 'POST'])]
+    public function partsEdit(
+        Request $request,
+        Vehicle $vehicle,
+        Part $part,
+        EntityManagerInterface $entityManager
+    ): Response {
+        $form = $this->createForm(PartType::class, $part);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $entityManager->flush();
+
+            return $this->redirectToRoute('vehicle_parts', [
+                'id' => $vehicle->getId(),
+            ], Response::HTTP_SEE_OTHER);
+        }
+
+        return $this->renderForm('vehicle/parts_edit.html.twig', [
+            'vehicle' => $vehicle,
+            'part' => $part,
+            'form' => $form,
+        ]);
+    }
+
+    #[Route('/{id}/parts/{partId}', name: 'vehicle_parts_delete', methods: ['POST'])]
```

Comment:
this is ambiguous: prefer `'/{vehicle}/parts/{part}'`to explicitly connect the first id to vehicle and the second one to part